src/read.rs

```diff
@@ -851,13 +865,44 @@ fn parse_escape<'de, R: Read<'de>>(read: &mut R, scratch: &mut Vec<u8>) -> Resul
 
```

Comment:
The documentation on deserialize_bytes says it permits lone surrogates, but this codepath is still returning error on a 0xDC00..=0xDFFF surrogate.